cluster-controller/src/main/java/io/strimzi/controller/cluster/model/AbstractModel.java

```diff
@@ -626,4 +627,39 @@ protected void kafkaHeapOptions(List<EnvVar> envVars, double dynamicHeapFraction
             envVars.add(buildEnvVar(ENV_VAR_KAFKA_HEAP_OPTS, trim));
         }
     }
+
+    public static void checkSingleNumber(Map<String, Object> data, String field) {
+        try {
+            if (data.get(field) == null) { // value is not set in config map -> will be used default value
+                return;
+            }
+            Integer.parseInt(data.get(field).toString());
+        } catch (NumberFormatException e)  {
+            String msg = " is corrupted.";
+            if (data.get(field).toString().length() == 0) {
+                msg = " is empty.";
+            }
+            throw new InvalidConfigMapException(field + msg);
+        }
+    }
+
+    public static void checkType(Map<String, Object> data, String field) {
```

Comment:
I don't understand the name `checkType`, surely it should be `checkStorage`? And of course the same comment about validate+get applies.